File: src/algorithms/sort/selection-sort/selection-sort.py
from __future__ import annotations


def find_smallest(arr: list[int]) -> int:
    smallest = arr[0]
    smallest_index = 0
    for i in range(1, len(arr)):
        if arr[i] < smallest:
            smallest = arr[i]
            smallest_index = i
    return smallest_index


# Selection Sort
def selection_sort(arr: list[int]) -> list[int]:
    new_arr = []
    for _ in range(len(arr)):
        smallest = find_smallest(arr)
        new_arr.append(arr.pop(smallest))
    return new_arr

File: src/algorithms/sort/selection_sort/selection_sort.c
#include "selection_sort.h"
#include <stddef.h>

void selection_sort(int arr[], int size) {
  for (int i = 0; i < size - 1; i++) {
    int min_index = i;

    for (int j = i + 1; j < size; j++) {

      if (arr[min_index] > arr[j]) {
        min_index = j;
      }
    }

    if (min_index != i) {
      int temp = arr[i];
      arr[i] = arr[min_index];
      arr[min_index] = temp;
    }
  }
}
